packages/multichain-account-service/src/MultichainAccountService.ts

```diff
@@ -395,52 +389,187 @@ export class MultichainAccountService {
+   * @returns The validated create wallet parameters.
    */
-  async createMultichainAccountWallet({
+  #getValidatedCreateWalletParams({
     mnemonic,
-  }: {
-    mnemonic: string;
-  }): Promise<MultichainAccountWallet<Bip44Account<KeyringAccount>>> {
+    password,
+    type,
+  }: CreateWalletParams): CreateWalletValidatedParams {
+    if (type === 'restore' && password && mnemonic) {
+      return {
+        password,
+        mnemonic,
+        type: 'restore',
+      };
+    }
+
+    if (type === 'import' && mnemonic) {
+      return { mnemonic, type: 'import' };
+    }
+
+    if (type === 'create' && password) {
+      return { password, type: 'create' };
+    }
+
+    throw new Error('Invalid create wallet parameters.');
+  }
+
+  /**
+   * Creates a new multichain account wallet by importing an existing mnemonic.
+   *
+   * @param mnemonic - The mnemonic to use to create the new wallet.
+   * @returns The new multichain account wallet.
+   */
+  async #createWalletByImport(
+    mnemonic: Uint8Array,
+  ): Promise<MultichainAccountWallet<Bip44Account<KeyringAccount>>> {
+    log(`Creating new wallet by importing an existing mnemonic...`);
     const existingKeyrings = this.#messenger.call(
       'KeyringController:getKeyringsByType',
       KeyringTypes.hd,
     ) as HdKeyring[];
 
-    const mnemonicAsBytes = mnemonicPhraseToBytes(mnemonic);
-
     const alreadyHasImportedSrp = existingKeyrings.some((keyring) => {
       if (!keyring.mnemonic) {
         return false;
       }
-      return areUint8ArraysEqual(keyring.mnemonic, mnemonicAsBytes);
+      return areUint8ArraysEqual(keyring.mnemonic, mnemonic);
     });
 
     if (alreadyHasImportedSrp) {
       throw new Error('This Secret Recovery Phrase has already been imported.');
     }
 
-    log(`Creating new wallet...`);
-
     const result = await this.#messenger.call(
       'KeyringController:addNewKeyring',
       KeyringTypes.hd,
       { mnemonic },
     );
 
-    const wallet = new MultichainAccountWallet({
+    // The wallet is ripe for discovery
+    return new MultichainAccountWallet({
       providers: this.#providers,
       entropySource: result.id,
       messenger: this.#messenger,
     });
+  }
+
+  /**
+   * Creates a new multichain account wallet by creating a new vault and keychain.
+   *
+   * @param password - The password to encrypt the vault with.
+   * @returns The new multichain account wallet.
+   */
+  async #createWalletByNewVault(
+    password: string,
+  ): Promise<MultichainAccountWallet<Bip44Account<KeyringAccount>>> {
+    log(`Creating new wallet by creating a new vault and keychain...`);
+    await this.#messenger.call(
+      'KeyringController:createNewVaultAndKeychain',
+      password,
+    );
+
+    const entropySourceId = (await this.#messenger.call(
+      'KeyringController:withKeyring',
+      { type: KeyringTypes.hd },
+      async ({ metadata }) => {
+        return metadata.id;
+      },
+    )) as string;
```

Comment:
I think this could be a method on its own, like `#getPrimaryEntropySourceId` maybe?

Also, we could make this slightly "less heavy" by relying on the `KeyringController:getState` and use the `state.keyrings` instead?

We usually use `withKeyring` for mutable operations (which involves a bunch of additional checks + a mutex lock).

WDYT?